compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ConvertConv2DToWinograd.cpp

```diff
@@ -291,28 +165,58 @@ class ConvertConvToWinograd final : public OpRewritePattern<ConvOp> {
   LogicalResult matchAndRewrite(ConvOp convOp,
                                 PatternRewriter &rewriter) const override {
 
-    bool isNchw;
-    if (!isValidConv2d(convOp, isNchw)) {
+    bool isNchwFchw;
+    if (!isValidConv2d(convOp, isNchwFchw)) {
       return failure();
     }
 
-    // Check that kernel has been constant folded (by validating rank = 3)
+    // Create winograd filter transform op.
     Value kernel = convOp.getInputs()[1];
     auto kernelType = cast<ShapedType>(kernel.getType());
     if (!kernelType) {
       return failure();
     }
-    Type elementType = kernelType.getElementType();
-    ArrayRef<int64_t> kernelShape = kernelType.getShape();
-    if (kernelShape.size() != 3) {
+    SmallVector<int64_t> kernelShape(kernelType.getShape());
+    if (llvm::any_of(kernelShape, ShapedType::isDynamic)) {
```

Comment:
Wondering if you need this check? The only restriction here is that the kernel should be 3x3 right? So won't the check below be sufficient?

Reply:
I mostly included this because the kernels are typically constants, so we don't have a strong reason to support dynamic channel dims right now. It will probably work, but I wanted to just enable constant shapes for now. I'll make an issue for tracking and I can come back to this later to enable it.